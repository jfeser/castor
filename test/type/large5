select([s1_acctbal, s1_name, n1_name, p1_partkey, p1_mfgr, s1_address,
           s1_phone, s1_comment],
     ahashidx(depjoin(select([min(p_size) as lo, max(p_size) as hi],
                        dedup(select([p_size], part))) as k1,
                select([range as k0], range(k1.lo, k1.hi))) as s0,
       select([s1_acctbal, s1_name, n1_name, p1_partkey, p1_mfgr, s1_address,
               s1_phone, s1_comment],
         filter((((r1_name = "") &&
                 ((p1_size = s0.k0) &&
                 (strpos(p1_type, "") =
                 ((strlen(p1_type) - strlen("")) + 1)))) &&
                ((ps1_supplycost = min_cost) &&
                ((r1_name = r_name) && (ps_partkey = ps1_partkey)))),
           orderby([s1_acctbal desc, n1_name, s1_name, p1_partkey],
             join(true,
               alist(dedup(
                       select([r_name, ps_partkey],
                         join((s_suppkey = ps_suppkey),
                           join((s_nationkey = n_nationkey),
                             join((n_regionkey = r_regionkey), nation, region),
                             supplier),
                           partsupp))) as k2,
                 select([r_name, ps_partkey, min(ps_supplycost) as min_cost],
                   join((((r_name = k2.r_name) && (ps_partkey = k2.ps_partkey))
                        && (s_suppkey = ps_suppkey)),
                     join((s_nationkey = n_nationkey),
                       join((n_regionkey = r_regionkey),
                         alist(select([n_nationkey, n_regionkey], nation) as s33,
                           atuple([ascalar(s33.n_nationkey),
                                   ascalar(s33.n_regionkey)],
                             cross)),
                         alist(select([r_regionkey, r_name], region) as s34,
                           atuple([ascalar(s34.r_regionkey), ascalar(s34.r_name)],
                             cross))),
                       alist(select([s_suppkey, s_nationkey], supplier) as s31,
                         atuple([ascalar(s31.s_suppkey),
                                 ascalar(s31.s_nationkey)],
                           cross))),
                     alist(select([ps_partkey, ps_suppkey, ps_supplycost],
                             partsupp) as s29,
                       atuple([ascalar(s29.ps_partkey), ascalar(s29.ps_suppkey),
                               ascalar(s29.ps_supplycost)],
                         cross))))),
               join((p1_partkey = ps1_partkey),
                 join((s1_suppkey = ps1_suppkey),
                   join((s1_nationkey = n1_nationkey),
                     join((n1_regionkey = r1_regionkey),
                       select([n_name as n1_name, n_nationkey as n1_nationkey,
                               n_regionkey as n1_regionkey],
                         alist(select([n_nationkey, n_name, n_regionkey],
                                 nation) as s35,
                           atuple([ascalar(s35.n_nationkey),
                                   ascalar(s35.n_name), ascalar(s35.n_regionkey)],
                             cross))),
                       select([r_name as r1_name, r_regionkey as r1_regionkey],
                         alist(select([r_regionkey, r_name], region) as s36,
                           atuple([ascalar(s36.r_regionkey), ascalar(s36.r_name)],
                             cross)))),
                     select([s_nationkey as s1_nationkey,
                             s_suppkey as s1_suppkey, s_acctbal as s1_acctbal,
                             s_name as s1_name, s_address as s1_address,
                             s_phone as s1_phone, s_comment as s1_comment],
                       alist(supplier as s32,
                         atuple([ascalar(s32.s_suppkey), ascalar(s32.s_name),
                                 ascalar(s32.s_address),
                                 ascalar(s32.s_nationkey), ascalar(s32.s_phone),
                                 ascalar(s32.s_acctbal), ascalar(s32.s_comment)],
                           cross)))),
                   select([ps_supplycost as ps1_supplycost,
                           ps_partkey as ps1_partkey, ps_suppkey as ps1_suppkey],
                     alist(select([ps_partkey, ps_suppkey, ps_supplycost],
                             partsupp) as s30,
                       atuple([ascalar(s30.ps_partkey), ascalar(s30.ps_suppkey),
                               ascalar(s30.ps_supplycost)],
                         cross)))),
                 select([p_size as p1_size, p_type as p1_type,
                         p_partkey as p1_partkey, p_mfgr as p1_mfgr],
                   alist(select([p_partkey, p_mfgr, p_type, p_size], part) as s28,
                     atuple([ascalar(s28.p_partkey), ascalar(s28.p_mfgr),
                             ascalar(s28.p_type), ascalar(s28.p_size)],
                       cross)))))))),
       0))
