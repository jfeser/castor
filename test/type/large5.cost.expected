(Ok 4454901029610857472)
